### PR TITLE
Move Ambassador CRD handling to resource processor

### PR DESCRIPTION
**Pull request by impl**
## Description
This change removes the `handle_k8s_crd` method of the resource fetcher, instead delegating that work to the `AmbassadorProcessor` class introduced in a previous refactor. The generated IR remains the same.

## Testing
I've run this through the Python test suite for validation.

## Tasks That Must Be Done
- [ ] Did you update CHANGELOG.md?
- [ ] Did you add or update tests?
- [ ] Did you update documentation?
- [ ] Were there any special dev tricks you had to use to work on this code efficiently?
  + [ ] Did you add them to DEVELOPING.md?
- [ ] Is this a build change?
  + [ ] If so, did you test on both Mac and Linux?

## Other
* If this is a documentation change for a particular release, please open the pull request against the release branch.

### CODE REVIEW

File: python/ambassador/config/resourcefetcher.py [hunk 1]
```diff
@@ -233,16 +227,16 @@ def parse_watt(self, serialization: str, finalize: bool=True) -> None:
...
+            # These objects have to be processed first, in order, as they depend
+            # on each other.
+            watt_k8s_keys = ['service', 'endpoints', 'secret', 'ingressclasses', 'ingresses']
 
-            # ...then handle Ambassador CRDs.
-            for key in CRDTypes:
+            # Then we add everything else to be processed.
+            watt_k8s_keys += watt_k8s.keys()
+
+            for key in dict.fromkeys(watt_k8s_keys):
```
Comment (kflynn) [inline python/ambassador/config/resourcefetcher.py:237]: Hmmmmm. So your comment on line 230 for `watt_k8s_keys` lists a sequence of keys that must be processed _in order_, then this line generates a dict (to deduplicate the keys, presumably)... so won't iterating over the dict shred the order you so carefully set up?

File: python/ambassador/config/resourcefetcher.py [hunk 2]
```diff
@@ -308,25 +302,16 @@ def handle_k8s(self, raw_obj: dict) -> None:
...
-
-        if obj.gvk.kind in CRDTypes:
-            handler = self.handle_k8s_crd
-            # self.handle_k8s_crd will do its own dup checking.
-            check_dup = False
-        else:
-            handler_name = f'handle_k8s_{obj.gvk.kind.lower()}'
-            # self.logger.debug(f"looking for handler {handler_name} for K8s {kind} {name}")
-            handler = getattr(self, handler_name, None)
+        handler_name = f'handle_k8s_{obj.gvk.kind.lower()}'
```
Comment (kflynn) [inline python/ambassador/config/resourcefetcher.py:305]: I kinda feel like there should be convenience accessors for `obj.gvk.kind` and `obj.gvk.kind.lower()`. 🙂  Probably a separate PR...

File: python/ambassador/config/resourcefetcher.py (hunk 1 above)
Comment (impl) [reply python/ambassador/config/resourcefetcher.py:237]: I will add a comment to clarify, but this is a basically a convenient way to get around the lack of a distinct `OrderedSet` type. As of [Python 3.7](https://mail.python.org/pipermail/python-dev/2017-December/151283.html), dictionaries are insertion-ordered:

```
Python 3.7.6 (default, Jan 19 2020, 22:34:52) 
[GCC 9.2.1 20200117] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> foo = ['bar', 'baz', 'quux', 'quux', 'baz']
>>> dict.fromkeys(foo)
{'bar': None, 'baz': None, 'quux': None}
>>> dict.fromkeys(foo)
{'bar': None, 'baz': None, 'quux': None}
>>> dict.fromkeys(foo)
{'bar': None, 'baz': None, 'quux': None}
>>> dict.fromkeys(foo)
{'bar': None, 'baz': None, 'quux': None}
```

File: python/ambassador/config/resourcefetcher.py (hunk 2 above)
Comment (impl) [reply python/ambassador/config/resourcefetcher.py:305]: This is actually a super small change so I'll throw it in here as well. Although there are actually almost no places where the `.lower()` is still used (other than generating the domain in the GVK itself), so maybe just `.kind`?

**Comment by impl** [comment]
> To confirm: `AmbassadorProcessor` is picking up all the resource types for which nothing special needs to happen, correct? Definitely looks cleaner there.
> 
> I had one other question about the ordering of dictionary keys -- looks good other than that, thanks!!

Indeed, it's just directly shimming those Kubernetes objects across to Ambassador resources. That's why the processor's `process` method is just one line of code.

**Pull request merged**